### DISCUSSION TITLE
Add ability to fill data when previewing emails

### DISCUSSION THREAD

**Discussion by satazor** [resend/react-email#580] [Ideas]
`react-email` offers a neat `next.js` app that allows us developers to preview emails. Most emails are dynamic in the sense that props are passed to fill in placeholders.

It would be nice if we could fill in the data / props so that we get the preview with said data. I was thinking on having something similar to https://storybook.js.org/docs/react/essentials/controls as part of the dev app. Something simpler could also suffice, like a field that would allow us to input JSON.

Thoughts? 🙏

**Comment by zenorocha** [comment]
This is very connected to @guillaumelachaud's idea on #371, which I really like.

Having something like Storybook's controls will be a great addition in terms of developer experience.

**Reply by villesau** [reply]
Out of curiosity, why not using Storybook as a backing engine for this altogether?

**Reply by bobylito** [reply]
I second that question :)

**Reply by bobylito** [reply]
@villesau I have shared an example that integrates react-email with storybook here : https://github.com/resendlabs/react-email/discussions/975
